### PR TITLE
TA-247 Respondent Nightly setup

### PR DESCRIPTION
**Pull request by phaniperla**
# Description

Respondent nightly Pipeline setup

https://tools.hmcts.net/jira/browse/TA-247

### CODE REVIEW

**Review by mounikahmcts — APPROVED** [review]

File: Jenkinsfile_nightly [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 
 properties([
-        pipelineTriggers([cron('00 21 * * *')]),
-        parameters([
-                string(name: 'URL_TO_TEST', defaultValue: 'https://respond-divorce-aks.aat.platform.hmcts.net', description: 'The URL you want to run these tests against'),
-        ])
+  pipelineTriggers([cron('00 14 * * *')]),
+  parameters([
+    string(name: 'URL_TO_TEST', defaultValue: 'https://respond-divorce-aks.aat.platform.hmcts.net', description: 'The URL you want to run these tests against'),
+  ])
```
Comment (Rob-Doherty) [inline Jenkinsfile_nightly:7]: Why do you want to run the nightly jobs at 2pm?

File: Jenkinsfile_nightly [hunk 2]
```diff
@@ -34,14 +35,27 @@ static LinkedHashMap<String, Object> secret(String secretName, String envVar) {
...
+  env.TEST_URL = params.URL_TO_TEST
+  env.NODE_ENV= 'ci'
+  env.test_environment = 'aat'
 
-    loadVaultSecrets(secrets)
-    enableCrossBrowserTest()
-    enableSecurityScan()
+  enableCrossBrowserTest()
+  enableSecurityScan()
+  loadVaultSecrets(secrets)
```
Comment (Rob-Doherty) [inline Jenkinsfile_nightly:44]: the vault secrets are needed in the `CrossBrowserTests`, so I think this should go back to the top of the test steps like it was originally:
```
    loadVaultSecrets(secrets)
    enableCrossBrowserTest()
    enableSecurityScan()
```

**Review by Rob-Doherty — CHANGES_REQUESTED** [review]

File: Jenkinsfile_nightly (hunk 1 above)
Comment (phaniperla) [reply Jenkinsfile_nightly:7]: When do you want me schedule  the corn job, 

updated corn job at 21:00

File: Jenkinsfile_nightly (hunk 2 above)
Comment (phaniperla) [reply Jenkinsfile_nightly:44]: updated

**Review by phaniperla — COMMENTED** [review]
Addressed reviewer comments

File: package.json [hunk 3]
```diff
@@ -23,7 +23,8 @@
     "test:coverage": "nyc check-coverage --statements 80 --branches 80 --functions 80 --lines 80",
     "test:a11y": "NODE_ENV=testing NODE_PATH=. LOG_LEVEL=ERROR mocha --exit test/a11y --reporter spec --recursive --timeout 15000 --exit --reporter mochawesome --reporter-options reportDir=functional-output,reportFilename=a11y",
     "sonar-scan": "node_modules/sonar-scanner/bin/sonar-scanner",
-    "test:functional": "NODE_PATH=. LOG_LEVEL=DEBUG codeceptjs run -c test/e2e/codecept.conf.js --steps --grep '@Pipeline' --reporter mochawesome",
+    "test:functional": "NODE_PATH=. LOG_LEVEL=DEBUG codeceptjs run-multiple parallel -c test/e2e/codecept.conf.js --steps --grep '@Pipeline' --reporter mocha-multi",
+    "test:fullfunctional": "NODE_PATH=. LOG_LEVEL=DEBUG codeceptjs run-multiple parallel -c test/e2e/codecept.conf.js --steps --grep '@Pipeline' --reporter mocha-multi",
```
Comment (Rob-Doherty) [inline package.json:27]: nice! 👌

**Review by Rob-Doherty — APPROVED** [review]
looks good to me 👍

File: package.json (hunk 3 above)
Comment (phaniperla) [reply package.json:27]: Thank you

**Pull request merged**